doc/PROTOCOL-SEMANTICS.md

```diff
@@ -0,0 +1,268 @@
+# gRPC semantics
+
+gRPC has many features, but the most fundamental core is the ability to perform
+RPCs. The goal of this document is defining the semantics of gRPC's RPCs.
+
+## Channels
+
+Most implementations will have Channel and Server concepts. A Channel is a
+virtual connection to an endpoint, capable of sending RPCs. Channel is "virtual"
+because the Channel is free to have zero or many actual connections. A Channel
+is free to determine which actual endpoint to use and may change it every RPC,
+permitting client-side load balancing. A Server is capable of receiving incoming
+connections and receiving RPCs.
+
+A "connection" is not a gRPC semantic concept and thus users should not assume a
+correlation between connections and RPCs. Although, practically, users should be
```

Comment:
Stickiness is not a thing at this level. It is possible for certain features to provide more guarantees. Note that stickiness _still_ doesn't provide any guarantee here, as we may get disconnected and re-connect between RPCs.

Yes, lots of users have asked for per-connection tracking, but this is generally in the context of authentication and they have been generally broken. The main problem is that the client won't know what transport it will get. Also, any per-connection tracking is broken when proxies are involved. So gRPC as a whole definitely doesn't offer this.